ratpack-core/src/main/java/ratpack/handling/internal/DefaultByContentSpec.java

```diff
@@ -90,7 +96,26 @@ public ByContentSpec noMatch(String mimeType) {
     return this;
   }
 
+  @Override
+  public ByContentSpec unspecified(Block handler) {
+    unspecifiedHandler = ctx -> handler.execute();
+    return this;
+  }
+
+  @Override
+  public ByContentSpec unspecified(String mimeType) {
+    unspecifiedHandler = ctx -> {
+      ctx.getResponse().contentType(mimeType);
+      blocks.get(mimeType).execute();
```

Comment:
Previously, it failed with a NullPointerException.  Now, it calls the ServerErrorHandler with a meaningful error message (and has tests to verify that behavior).  I think that it's the right thing, in this case, to fail at runtime, because doing the check at configuration-time would require the block to be registered before the noMatch/unspecified, and I don't think we should require those declarations to be in a particular order.